cpp/src/io/comp/uncomp.cpp

```diff
@@ -560,5 +560,69 @@ size_t decompress(compression_type compression,
   }
 }
 
+size_t estimate_uncompressed_size(compression_type compression, host_span<uint8_t const> src)
```

Comment:
Is this an estimate or the exact size?
Have you measured the performance impact from the additional work we do to get the decompressed size?

Reply:
Good point, this function returns the exact size. I'll change the name to `get_uncompressed_size`.
I'll post the performance impact of this change shortly.